### PR TITLE
[ldm3d] Ldm3d upscaler to community pipeline

### PR DESCRIPTION
**Pull request by estelleafl**
Hi @patrickvonplaten,
Finally, I created a new PR for our contribution to the community pipelines. We can close the previous one (#5660)
Thank you for your help
Best

### CODE REVIEW

File: src/diffusers/image_processor.py [hunk 1]
```diff
@@ -441,6 +450,21 @@ def numpy_to_pil(images: np.ndarray) -> List[PIL.Image.Image]:
...
+        """
+        Convert a PIL image or a list of PIL images to NumPy arrays.
+        """
+        if not isinstance(images, list):
+            images = [images]
+
+        images = [np.array(image).astype(np.float32) / (2**16 - 1) for image in images]
+        images = np.stack(images, axis=0)
+        # images = images *2 -1
+        # images = (images * 127.5 + 127.5)/255
```
Comment (patrickvonplaten) [inline src/diffusers/image_processor.py:465]:
```suggestion
```

File: src/diffusers/image_processor.py [hunk 2]
```diff
@@ -441,6 +450,21 @@ def numpy_to_pil(images: np.ndarray) -> List[PIL.Image.Image]:
 
         return pil_images
 
+    @staticmethod
+
```
Comment (patrickvonplaten) [inline src/diffusers/image_processor.py:454]:
```suggestion
```

File: src/diffusers/image_processor.py [hunk 3]
```diff
@@ -523,6 +548,105 @@ def postprocess(
             return image[:, :, :, :3], image_depth
 
         if output_type == "pil":
-            return self.numpy_to_pil(image), self.numpy_to_depth(image)
+            return self.numpy_to_pil(image), self.numpy_to_pil_depth(image)
         else:
             raise Exception(f"This type {output_type} is not supported")
+
+    def preprocess(
```
Comment (patrickvonplaten) [inline src/diffusers/image_processor.py:555]: Hmm is this really needed?

File: src/diffusers/image_processor.py [hunk 4]
```diff
@@ -452,7 +476,8 @@ def rgblike_to_depthmap(image: Union[np.ndarray, torch.Tensor]) -> Union[np.ndar
         """
         return image[:, :, 1] * 2**8 + image[:, :, 2]
 
-    def numpy_to_depth(self, images: np.ndarray) -> List[PIL.Image.Image]:
+    def numpy_to_pil_depth(self, images: np.ndarray) -> List[PIL.Image.Image]:
```
Comment (patrickvonplaten) [inline src/diffusers/image_processor.py:479]: Can we try to leave the name as is - otherwise this would break backwards compatibility

Comment (estelleafl) [reply src/diffusers/image_processor.py:479]: sounds good

File: src/diffusers/image_processor.py (hunk 3 above)
Comment (estelleafl) [reply src/diffusers/image_processor.py:555]: the preprocess? or the raise?

Comment (patrickvonplaten) [reply src/diffusers/image_processor.py:552]: Ah sorry I misread here, actually this look ok to me - @yiyixuxu wdyt?

**Review by patrickvonplaten — APPROVED** [review]
The changes look good to me - @yiyixuxu wdyt?

File: src/diffusers/image_processor.py (hunk 3 above)
Comment (yiyixuxu) [reply src/diffusers/image_processor.py:552]: yeah fine with me too!

**Comment by yiyixuxu** [comment]
> I am getting the error below when running make style/make quality. I tried also from main and not on my branch and Im getting this error as well. Would you know why?
> 
> ruff check examples scripts src tests utils setup.py --fix error: TOML parse error at line 16, column 1 | 16 | [tool.ruff.format] | ^^^^^^^^^^^^^^^^^^ wanted exactly 1 element, more than 1 element
> 
> make: *** [Makefile:56: style] Error 2

can you try upgrading ruff?

```
pip install --upgrade ruff
```

**Comment by yiyixuxu** [comment]
@estelleafl 
merged! thank you :)

**Pull request merged**